### PR TITLE
Forward Content-Encoding header OKAPI-936

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Fix for OKAPI-936 Request logging can send compressed data to the filter. Until now only the Content-Type was forwarded from request-response pipeline.. request-only, headers should NOT forward this as the response is thrown away.

### CODE REVIEW

File: okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java [hunk 1]
```diff
@@ -844,21 +850,24 @@ private void proxyRequestResponse(Iterator<ModuleInstance> it,
...
           relayToRequest(res, pc, mi);
-          final String ct = res.getHeader("Content-Type");
-          if (ct != null) {
-            ctx.request().headers().set("Content-Type", ct);
+          for (String header : FORWARD_HEADERS) {
+            String value = res.getHeader(header);
+            request.headers().remove(header);
+            if (value != null) {
+              request.headers().set(header, value);
+            }
```
Comment (hjiebsco) [inline okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java:864]: Seems you can combine those to one line `request.headers().set(header, res.getHeader(header));`

File: okapi-core/src/test/java/org/folio/okapi/ProxyTest.java [hunk 2]
```diff
@@ -4344,4 +4354,170 @@ public void testManyModules(TestContext context) throws IOException {
...
+    Assert.assertTrue(
+        "raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+
+    given().header("X-Okapi-Tenant", okapiTenant)
+        .header("Content-Type", "text/plain")
+        .body("Okapi").post("/echo")
+        .then().statusCode(200)
+        .header("Content-Type", "text/plain; charset=ISO-8859-1")
+        .body(equalTo("Okapi"));
```
Comment (hjiebsco) [inline okapi-core/src/test/java/org/folio/okapi/ProxyTest.java:4476]: Consider to add a line to check non-existing gzip header `.header("Content-Encoding", nullValue())`

**Review by hjiebsco — APPROVED** [review]
Left two minor comments.

File: okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java (hunk 1 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java:864]: Can I? Setting null is the same as not setting it at all?

File: okapi-core/src/test/java/org/folio/okapi/ProxyTest.java (hunk 2 above)
Comment (adamdickmeiss) [reply okapi-core/src/test/java/org/folio/okapi/ProxyTest.java:4476]: ok

File: okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java (hunk 1 above)
Comment (hjiebsco) [reply okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java:864]: The doc says `If there is an existing header with the same name, it is removed. Setting a null value removes the entry.` So I think it is yes. Please verify though.

Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java:864]: Right you are.. Thx

**Review by hjiebsco — APPROVED** [review]

**Pull request merged**